packages/browser-wallet/src/popup/pages/AddWeb3IdCredential/AddWeb3IdCredential.tsx

```diff
@@ -144,18 +146,20 @@ export default function AddWeb3IdCredential({ onAllow, onReject }: Props) {
         const credentialHolderId = wallet.getVerifiableCredentialPublicKey(issuer, index).toString('hex');
         const credentialSubjectId = createPublicKeyIdentifier(credentialHolderId, network);
         const credentialSubject = { ...credential.credentialSubject, id: credentialSubjectId };
+        const credentialId = createCredentialId(credentialHolderId, issuer, network);
 
         const fullCredential = {
             ...credential,
             credentialSubject,
-            id: createCredentialId(credentialHolderId, issuer, network),
+            id: credentialId,
             index,
         };
         await setWeb3IdCredentials([...web3IdCredentials.value, fullCredential]);
         if (metadata) {
             const newMetadata = { ...verifiableCredentialMetadata.value };
             newMetadata[metadataUrl.url] = metadata;
             await setVerifiableCredentialMetadata(newMetadata);
+            await setTempMetadata({ ...tempMetdata.value, [credentialId]: metadata });
```

Comment:
This is probably fine for now, but I think we could avoid duplicating the metadata. 

The problem is that we can't find the `credentialEntry` so the metadata hook could also look somewhere else for the metadata URL, which should be all that is necessary to store to fix problem.